src/apdu_sign.c

```diff
@@ -586,22 +581,37 @@ void output_end(void) {
                 memcpy(G.u.tx.outputs[0].destination.hash, dest_to_show, 20);
             }
         } else if(G.cell_state.lock_arg_nonequal) {
-            if((G.maybe_transaction.v.flags & HAS_DESTINATION_ADDRESS) && memcmp(G.u.tx.outputs[0].destination.hash, G.lock_arg_tmp.hash, 20)) {
-                if(G.maybe_transaction.v.tag != OPERATION_TAG_NOT_SET && G.maybe_transaction.v.tag != OPERATION_TAG_MULTI_OUTPUT_TRANSFER)
+            // if the output lock arg doesn't match the change bip-32 path
+            if (!(G.maybe_transaction.v.flags & HAS_DESTINATION_ADDRESS) ) {
+                if (G.u.tx.output_count != 0) {
```

Comment:
Do we need this check?

Reply:
It's supposed to be sanity check assertion. I would use an `assert(..)` if we had one to make that clearer.